File: src/modelo/produto.ts
export default class Produto {
    public nome!: string
    public preco!: string
    public marca!: string
    public genero!: string

    constructor(nome: string,  marca: string, preco: string, genero: string){
        this.nome = nome
        this.marca = marca
        this.preco = preco
        this.genero = genero
    }
    public atualizarDados(nome: string, marca: string, preco: string, genero: string): void {
        this.nome = nome;
        this.marca = marca;
        this.preco = preco;
        this.genero = genero;
    }
    public get getPreco() {
        return this.preco
    }
    
}

File: wbbackend/src/main/java/com/wb/backend/entidades/Produto.java
package com.wb.backend.entidades;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import org.springframework.hateoas.RepresentationModel;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Produto extends RepresentationModel<Produto> {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column
	private String nome;

	@Column
	private String marca;

	@Column
	private String preco;

	public Produto(){
	}

	public Produto(String nome, String marca, String preco){
		this.nome = nome;
		this.marca = marca;
		this.preco = preco;
	}
}